### PR TITLE
update the-hard-way.md

### PR DESCRIPTION
**Pull request by cmssczy**
### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

some infomation is outdated in the-hard-way.md 

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: docs/en/latest/practices/the-hard-way.md [hunk 1]
```diff
@@ -730,21 +737,22 @@ After the ingress controller status is converted to `Running`, we could create a
...
-        servicePort: 80
+  - name: httpbin
+    match:
+      hosts:
+      - local.httpbin.org
+      paths:
+      - /*
+    backends:
+       - serviceName: httpbin
+         servicePort: 80
```
Comment (tao12345666333) [inline docs/en/latest/practices/the-hard-way.md:755]:
```suggestion
    backends:
      - serviceName: httpbin
        servicePort: 80
```

Comment (tao12345666333) [reply docs/en/latest/practices/the-hard-way.md:755]: remove a space

**Review by tao12345666333 — COMMENTED** [review]
Thanks！

File: docs/en/latest/practices/the-hard-way.md (hunk 1 above)
Comment (cmssczy) [reply docs/en/latest/practices/the-hard-way.md]: Done

File: docs/en/latest/practices/the-hard-way.md [hunk 2]
```diff
@@ -127,6 +127,10 @@ spec:
           volumeMounts:
             - name: data
               mountPath: /etcd
+      # If you don't have a storage provisioner or don't want to use persistence volume, you could use an `emptyDir` as follow.
+      # volumes:
+      #   - name: data
+      #     emptyDir: {}
```
Comment (tao12345666333) [inline docs/en/latest/practices/the-hard-way.md:133]:
```suggestion
      #     emptyDir: {}
```

Comment (tao12345666333) [reply docs/en/latest/practices/the-hard-way.md:133]: need remove space :joy_cat:

Comment (cmssczy) [reply docs/en/latest/practices/the-hard-way.md]: done

**Review by tao12345666333 — APPROVED** [review]
thanks

**Comment by tao12345666333** [comment]
ping @lingsamuel @tokers @gxthrj  for review. Thanks!

File: docs/en/latest/practices/the-hard-way.md [hunk 3]
```diff
@@ -443,11 +447,13 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: apisix-ingress-controller
+  namespace: apisix
```
Comment (lingsamuel) [inline docs/en/latest/practices/the-hard-way.md:450]: Do we need to hardcode the namespace? It is already configured in the command line.

Comment (cmssczy) [reply docs/en/latest/practices/the-hard-way.md:450]: This doc tells us to install apisix-ingress-controller manually but not by helm. And all the configurations are all deployed in `apisix` namespace, deployment/serviceAccount/clusterRole, etc. So I think we should specify this parameter explicitly.

Comment (tao12345666333) [reply docs/en/latest/practices/the-hard-way.md:450]: I agree, we can keep them consistent

**Review by lingsamuel — APPROVED** [review]

**Review by gxthrj — APPROVED** [review]

**Pull request merged**